app/src/main/java/io/homeassistant/companion/android/settings/qs/views/ManageTilesView.kt

```diff
@@ -0,0 +1,144 @@
+                fontSize = 15.sp
+            )
+        }
+        DropdownMenu(expanded = expandedTile, onDismissRequest = { expandedTile = false }) {
+            val tileNameArray = stringArrayResource(id = io.homeassistant.companion.android.R.array.tile_name)
+            val tileIdArray = stringArrayResource(id = io.homeassistant.companion.android.R.array.tile_ids)
+            for ((tileName, tileId) in tileNameArray.zip(tileIdArray)) {
+                DropdownMenuItem(onClick = {
+                    viewModel.selectedTile.value = tileId
+                    expandedTile = false
+                    if (viewModel.currentTile() != null) {
+                        viewModel.tileLabel.value = viewModel.currentTile()!!.label
+                        viewModel.tileSubtitle.value = viewModel.currentTile()!!.subtitle
+                        viewModel.selectedEntityId.value = viewModel.currentTile()!!.entityId
+                        viewModel.selectedIcon.value = viewModel.currentTile()!!.iconId
+                        val loader = IconPackLoader(context)
+                        iconPack = createMaterialDesignIconPack(loader)
+                        iconPack.loadDrawables(loader.drawableLoader)
+                        val iconDrawable = viewModel.selectedIcon.value?.let { iconPack.getIcon(it)?.drawable }
+                        if (iconDrawable != null) {
+                            val icon = DrawableCompat.wrap(iconDrawable)
+                            icon.setColorFilter(context.resources.getColor(io.homeassistant.companion.android.R.color.colorAccent), PorterDuff.Mode.SRC_IN)
+                            viewModel.icon.value = icon
+                        }
+                    }
+                }) {
+                    Text(tileName)
+                }
+            }
+        }
+        Text(
+            text = stringResource(id = R.string.tile_label),
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        TextField(
+            value = viewModel.tileLabel.value.toString(),
+            onValueChange = { viewModel.tileLabel.value = it },
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        Text(
+            text = stringResource(id = R.string.tile_subtitle),
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+        TextField(
+            value = viewModel.tileSubtitle.value.toString(),
+            onValueChange = { viewModel.tileSubtitle.value = it },
+            modifier = Modifier.padding(start = 10.dp, bottom = 10.dp)
+        )
+
+        TextButton(onClick = {
+            iconDialog.show(childFragment, viewModel.selectedTile.value)
+        }) {
+            Text(
+                text = stringResource(id = R.string.tile_icon),
+                fontSize = 15.sp,
+                modifier = Modifier.padding(end = 10.dp)
+            )
+            viewModel.icon.value?.toBitmap()?.asImageBitmap()

```

Comment:
A color filter should be added because in light mode the default icon is white

Reply:
Just submitted a change and I thought it would apply here but for some reason its not applying to the default icon. Not sure what I did wrong here :)

Reply:
It looks like you only apply the tint to the `selectedIcon` but the default icon is the default value for `drawableIcon` (which is never updated because `selectedIcon` is not set).  You'd have to move the filter to always tint the drawable icon in the view model, or do it in Compose on the `Image`: `colorFilter = ColorFilter.tint(colorResource(R.color.colorIcon))`.

Reply:
much cleaner to do it in compose :) thanks for the help!!!